apps/heft/src/cli/actions/BuildAction.ts

```diff
@@ -6,15 +6,96 @@ import {
   CommandLineStringParameter,
   ICommandLineActionOptions
 } from '@rushstack/ts-command-line';
+import { SyncHook, AsyncParallelHook, AsyncSeriesHook } from 'tapable';
+import { performance } from 'perf_hooks';
 
 import { HeftActionBase, IHeftActionBaseOptions, ActionHooksBase, IActionDataBase } from './HeftActionBase';
+import { CleanAction } from './CleanAction';
 
-export interface IBuildActionOptions extends IHeftActionBaseOptions {}
+export interface IBuildActionOptions extends IHeftActionBaseOptions {
+  cleanAction: CleanAction;
+}
+
+/**
+ * @public
+ */
+export class BuildPhaseHooksBase {
```

Comment:
Is "phase" a good name for this? 

In our discussion with @D4N14L a couple months ago, we sketched out a design for a "multiphase" feature that gives Rush more fine-grained control over the toolchain operations for each project.  (I keep forgetting to create a GitHub issue with the design note.)

In that model, the word "phase" means a named operation that Rush can tell a project's toolchain to perform. The dependency topology for phases is shared across all projects in the monorepo, and declared in the **command-line.json** config file:

```
    {
      "commandKind": "multiphase", # or phased?
      "name": "build",
      "safeForSimultaneousRushProcesses": false,
      "phases": [ "buildPhase", "testPhase", "docsPhase" ] # Order not important, as this will be determined by the phase configuration. All required phases should be specified for clarity, throw error if not true
      # Other properties have been moved into the individual phases
    }
```

If we implement that feature, then it might be confusing for Heft to refer to its operations as "phases" if they are something different.

Reply:
To avoid confusion, maybe Heft's phases should be renamed to "steps" or "stages" or somesuch.

Reply:
Agreed on the potential confusion. What about "build stage?"

Reply:
We decided on "stage". And then later if we need to formalize operations within a stage, those would be "steps"